### PR TITLE
OCPBUGS-17436: Enable DynamicResourceAllocation API in kube-apiserver

### PR DESCRIPTION
**Pull request by bertinatto**
/assign @benluddy
CC @ingvagabund @soltysh

### CODE REVIEW

File: pkg/operator/configobservation/apienablement/observe_runtime_config.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ import (
 
 var DefaultGroupVersionsByFeatureGate = map[configv1.FeatureGateName][]schema.GroupVersion{
 	"ValidatingAdmissionPolicy": {{Group: "admissionregistration.k8s.io", Version: "v1alpha1"}},
+	"DynamicResourceAllocation": {{Group: "resource.k8s.io", Version: "v1alpha2=true"}},
```
Comment (benluddy) [inline pkg/operator/configobservation/apienablement/observe_runtime_config.go:19]:
```suggestion
	"DynamicResourceAllocation": {{Group: "resource.k8s.io", Version: "v1alpha2"}},
```

**Comment by benluddy** [comment]
/hold

To allow time for a smoke test since there is no tech preview presubmit.

**Comment by bertinatto** [comment]
TechPreview cluster installed successfully:

```
~ ❯ oc get clusterversion
NAME      VERSION                                                   AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         3m33s   Cluster version is 4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest

~ ❯ oc get co
NAME                                       VERSION                                                   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
authentication                             4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      4m8s
baremetal                                  4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
cloud-controller-manager                   4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      28m
cloud-credential                           4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      28m
cluster-api                                4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
cluster-autoscaler                         4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
config-operator                            4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      25m
console                                    4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      6m49s
control-plane-machine-set                  4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      21m
csi-snapshot-controller                    4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
dns                                        4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
etcd                                       4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      23m
image-registry                             4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      11m
ingress                                    4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      12m
insights                                   4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      18m
kube-apiserver                             4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      7m4s
kube-controller-manager                    4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      23m
kube-scheduler                             4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      22m
kube-storage-version-migrator              4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      25m
machine-api                                4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      10m
machine-approver                           4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
machine-config                             4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      3m25s
marketplace                                4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      23m
monitoring                                 4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      3m59s
network                                    4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      26m
node-tuning                                4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
olm                                        4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      10m
openshift-apiserver                        4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      7m9s
openshift-controller-manager               4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      24m
openshift-samples                          4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      17m
operator-lifecycle-manager                 4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      23m
operator-lifecycle-manager-catalog         4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      23m
operator-lifecycle-manager-packageserver   4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      18m
platform-operators-aggregated              4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      7m9s
service-ca                                 4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      25m
storage                                    4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest   True        False         False      11m

~ ❯ oc get featuregates cluster -o yaml
apiVersion: config.openshift.io/v1
kind: FeatureGate
metadata:
  creationTimestamp: "2023-08-21T17:21:53Z"
  generation: 1
  name: cluster
  resourceVersion: "1358"
  uid: c9df204a-0ef8-4499-aaab-c88209ede53d
spec:
  featureSet: TechPreviewNoUpgrade
status:
  featureGates:
  - disabled:
    - name: EventedPLEG
    - name: MachineAPIOperatorDisableMachineHealthCheckController
    enabled:
    - name: AdminNetworkPolicy
    - name: AdmissionWebhookMatchConditions
    - name: AlibabaPlatform
    - name: AutomatedEtcdBackup
    - name: AzureWorkloadIdentity
    - name: BuildCSIVolumes
    - name: CSIDriverSharedResource
    - name: CloudDualStackNodeIPs
    - name: DynamicResourceAllocation
    - name: ExternalCloudProvider
    - name: ExternalCloudProviderAzure
    - name: ExternalCloudProviderExternal
    - name: ExternalCloudProviderGCP
    - name: GCPLabelsTags
    - name: GatewayAPI
    - name: InsightsConfigAPI
    - name: MachineAPIProviderOpenStack
    - name: MaxUnavailableStatefulSet
    - name: NodeSwap
    - name: OpenShiftPodSecurityAdmission
    - name: PrivateHostedZoneAWS
    - name: RetroactiveDefaultStorageClass
    - name: RouteExternalCertificate
    - name: SigstoreImageVerification
    - name: VSphereStaticIPs
    - name: ValidatingAdmissionPolicy
    version: 4.14.0-0.ci.test-2023-08-21-170954-ci-ln-thn79sk-latest
```

/hold cancel

**Pull request merged**